community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexMapReference.java

```diff
@@ -41,16 +42,26 @@ public IndexProxy getIndexProxy( long indexId ) throws IndexNotFoundKernelExcept
         return proxy;
     }
 
-    public IndexProxy getOnlineIndexProxy( long indexId ) throws IndexNotFoundKernelException
+    public IndexProxy getIndexProxy( IndexDescriptor descriptor ) throws IndexNotFoundKernelException
     {
-        IndexProxy proxy = getIndexProxy( indexId );
+        IndexProxy proxy = indexMap.getIndexProxy( descriptor );
+        if ( proxy == null )
+        {
+            throw new IndexNotFoundKernelException( "No index for " + descriptor + " exists." );
+        }
+        return proxy;
+    }
+
+    public IndexProxy getOnlineIndexProxy( IndexDescriptor descriptor ) throws IndexNotFoundKernelException
+    {
+        IndexProxy proxy = getIndexProxy( descriptor );
         switch ( proxy.getState() )
         {
             case ONLINE:
                 return proxy;
 
             default:
-                throw new IndexNotFoundKernelException( "Expected index with id " + indexId + " to be online.");
+                throw new IndexNotFoundKernelException( "Expected index on " + descriptor + " to be online.");
```

Comment:
It would be nice if we spelled out behaviour for the other values of the `InternalIndexState` enum, to signal that they are not forgotten, but intended to fall into this default case.